django/forms/widgets.py

```diff
@@ -979,7 +979,11 @@ def get_context(self, name, value, attrs):
         date_context['year'] = self.select_widget(attrs, choices=year_choices).get_context(
             name=year_name,
             value=context['widget']['value']['year'],
-            attrs={**context['widget']['attrs'], 'id': 'id_%s' % year_name},
+            attrs={
+                **context['widget']['attrs'],
+                'id': 'id_%s' % year_name,
+                'placeholder': _('Year') if self.is_required else False
```

Comment:
In each of the three cases, can you change `False` to `None` and add a trailing comma.

(I think we *could* also move the placeholder line to the top to allow it to be overridden by the context, but it may not be worth it as I'm not sure the attributes passed down can be specified for each select individually...)

Reply:
If I change `False` to `None`, rendered HTML would contain `placeholder="None"` when `required=False`.
I think we should keep `False`.

Reply:
Does it not render with `placeholder="False"` then? Seems strange to me...